src/lib/Libwin/util.c

```diff
@@ -951,32 +910,41 @@ impersonate_user(HANDLE hlogintoken)
 {
 	is_user_impersonated = FALSE;
 
-	if (hlogintoken == NULL || hlogintoken == INVALID_HANDLE_VALUE)
+	if (hlogintoken == NULL || hlogintoken == INVALID_HANDLE_VALUE) {
+		errno = ERROR_INVALID_PARAMETER;
+		SetLastError(ERROR_INVALID_PARAMETER);
 		return FALSE;
+	}
 
 	is_user_impersonated = ImpersonateLoggedOnUser(hlogintoken);
 	if (is_user_impersonated == FALSE)
 		errno = GetLastError();
+	/* error logging to be done by the caller */
 	return is_user_impersonated;
 }
 
 /* @brief
- *  	Revert an impersonated user, a wrapper to RevertToSelf() API
+ *  	Revert an impersonated user, an error loggin wrapper to RevertToSelf() API
  *  	If revert fails, sets the errno.
+ * 
+ * @param[in]   funcname : function name where API was invoked
+ * @param[in]   lineno : line number of API invocation
  *
  *  @return     BOOL
  *  @retval     whether revert succeded, return value of RevertToSelf() API
  */
 BOOL
-revert_impersonated_user()
+_log_wrap_revert_impersonated_user(LPCSTR funcname, INT lineno)
```

Comment:
Looks like we are removing "revert_impersonated_user" then why this function name is "_log_wrap_revert_impersonated_user" ?

Reply:
Only the definition header is renamed, so as to be helpful with macro replacement, the invocation will still be called as `revert_impersonated_user`. This is to get the function name and line location of invocation of `revert_impersonated_user` for error logging. The `_log_wrap_` is as per convention followed for wrapping `CloseHandle` and `LocalFree` in win.h ,  hit +1 if you agree.